http_unix_daemon.pl

```diff
@@ -447,13 +446,28 @@
 	option(certfile(CertFile), Options, 'https/server.crt'),
 	option(keyfile(KeyFile), Options, 'https/server.key'),
 	option(cipherlist(CipherList), Options, 'DEFAULT'),
-	prepare_https_certificate(CertFile, KeyFile),
-	SSL = ssl([ certificate_file(CertFile),
+	prepare_https_certificate(CertFile, KeyFile, Passwd0),
+	(   string(Passwd0)
+	->  Passwd = Passwd0
+	;   options_password(Options, Passwd)
+	),
+	file_to_string(CertFile, Certificate),
+	private_key(KeyFile, Passwd, Key),
+	SSL = ssl([ certificate(Certificate),
 		    cipher_list(CipherList),
-		    key_file(KeyFile),
-		    pem_password_hook(http_unix_daemon:ssl_passwd)
+		    key(Key)
 		  ]).
 
+file_to_string(File, String) :-
```

Comment:
Please use read_file_to_string/3 instead.  I wondered about the encoding to use, but the default text encoding is probably correct.